account_fiscal_position_partner_type/models/account_fiscal_position.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021 Valentin Vinagre <valentin.vinagre@sygel.es>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class AccountFiscalPosition(models.Model):
+    _inherit = "account.fiscal.position"
+
+    fiscal_position_type = fields.Selection(
+        selection=[("b2c", "End customer (B2C)"), ("b2b", "Company (B2B)")],
+        string="Type",
+        default=lambda self: self._default_fiscal_position_type(),
+    )
+
+    def _selection_fiscal_position_type(self):
+        field = "fiscal_position_type"
+        return self.env["account.fiscal.position"].fields_get(allfields=[field])[field][
+            "selection"
+        ]
+
+    @api.model
+    def _default_fiscal_position_type(self):
+        return self.env.company.default_fiscal_position_type
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        args = expression.AND(
+            (
+                args,
+                [
+                    (
+                        "fiscal_position_type",
+                        "=",
+                        self.env.context.get("fiscal_position_type", False),
```

Comment:
Only apply this if `"fiscal_position_type" in self.env.context`

Reply:
hummm if I apply it only when the 'fiscal_position_type' is in self.env.context. When the partner doesn't have a 'tax_position_type' configured, the search takes into account the tax position configured with 'tax_position_type' and it shouldn't, that's why I do it this way.

Reply:
Then always pass the context when coming from the other method, but what we don't want is that this limited search restricts other regular searches. If not, you won't be able to see all the fiscal position types. Do you understand me?

Reply:
Now yes :). Sorry, hard day.